### PR TITLE
Create stg classes

### PR DESCRIPTION
**Pull request by RJO23**
- Cleaned text.
- Wrote standard, unit and accepted values tests.
- Calculated class duration.

### CODE REVIEW

**Comment by RJO23** [comment]
https://github.com/orgs/cbfacademy/projects/46/views/1?pane=issue&itemId=171194128&issue=cbfacademy%7Cfittrack_analytics%7C5

**Review by rosie-taylor — DISMISSED** [review]
Could you add a unit test to this? Something that double checks edge cases (e.g. for duration)

File: models/staging/stg_classes.sql [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        cast(scheduled_end as timestamp) as end_time,
+
+        -- Calculating duration_minutes
+        timestamp_diff(
+            cast(scheduled_end as timestamp), 
+            cast(scheduled_start as timestamp), 
+            minute
+        ) as duration_minutes
+
+    from source
```
[the remark this answers is not in the document]
Comment (RJO23) [reply models/staging/stg_classes.sql:29]: I think it would be better to save ordering until later down the line?

**Review by EmilyYoung26 — APPROVED** [review]
Looks nice and clean and I can see you've made all the changes and added a singular test similar to my one. Nice job!

File: models/staging/stg_classes.sql (hunk 1 above)
[the remark this answers is not in the document]
Comment (harjeetkalsi) [reply models/staging/stg_classes.sql:29]: later down the line is better rather than in stagging

File: tests/singular_tests/stg_classes_start_time_before_end_time.sql [hunk 2]
```diff
@@ -0,0 +1,9 @@
+-- This test looks for any rows where the start_time is logically impossible (i.e., it happens after the end_time).
+-- If this query returns any rows, the dbt test will fail.
+
+select
+    class_id,
+    start_time,
+    end_time
```
Comment (harjeetkalsi) [inline tests/singular_tests/stg_classes_start_time_before_end_time.sql:7]: The column names don't seem to match the column names actually used. 

We have 'scheduled_start' and 'scheduled_end' i think, but double check

File: models/staging/stg_classes.sql [hunk 3]
```diff
@@ -5,13 +5,13 @@ with source as (
...
     select
         
-        cast(class_id as string) as class_id,
+        cast(class_id as integer) as class_id,
         
-        cast(instructor_id as string) as instructor_id,
+        cast(instructor_id as integer) as instructor_id,
         
-        cast(category_id as string) as category_id,
+        cast(category_id as integer) as category_id,
```
Comment (harjeetkalsi) [inline models/staging/stg_classes.sql]: One question to you hear: do we actually need to cast these IDs to integers? 
Since class_id, instructor_id, and category_id are identifiers rather than values we calculate with, I think keeping them as strings could be a valid and possibly safer modelling choice. Happy with either approach, but why are we casting these IDs to integers? is there a benefit beyond “they look numeric”?

File: models/staging/stg_classes.yml
Comment (harjeetkalsi) [inline models/staging/stg_classes.yml:54]: I might be missing this on my view, but where is the scheduled_end and scheduled_start columns and their not_null tests?

File: models/staging/stg_classes.yml [hunk 4]
```diff
@@ -14,19 +14,25 @@ models:
...
+              field: category_id
 
       - name: class_name
         description: "The cleaned name of the fitness class."
         tests:
           - not_null
           - accepted_values:
               arguments:
-                values: ['Boxing Basics', 'Boxing Conditioning', 'Evening Boxing', 'Express HIIT', 'Flow Yoga', 'HIIT Blast', 'Lunch Yoga', 'Mobility Reset', 'Morning Muay Thai', 'Morning Pilates', 'Muay Thai Advanced', 'Muay Thai Fundamentals', 'Pilates Core', 'Restore Yoga', 'Stretch & Recover']
+                values: ['Express Hiit', 'Hiit Blast', 'Boxing Basics', 'Boxing Conditioning', 'Evening Boxing', 'Express HIIT', 'Flow Yoga', 'HIIT Blast', 'Lunch Yoga', 'Mobility Reset', 'Morning Muay Thai', 'Morning Pilates', 'Muay Thai Advanced', 'Muay Thai Fundamentals', 'Pilates Core', 'Restore Yoga', 'Stretch & Recover']
```
Comment (harjeetkalsi) [inline models/staging/stg_classes.yml]: The class_name accepted values should match the actual cleaned output of initcap(trim(class_name)). Right now it looks like a mix of raw and cleaned values, especially around HIIT/Hiit. For example, because initcap would turn:

HIIT Blast into Hiit Blast
Express HIIT into Express Hiit 

We shouldnt have the raw version in the accepted values as we are checking they turned into the clean version.

**Review by harjeetkalsi — CHANGES_REQUESTED** [review]
This is looking good, just a few changes and things to think about.

File: models/staging/stg_classes.sql (hunk 3 above)
Comment (RJO23) [reply models/staging/stg_classes.sql]: I agree, however I was getting an error stating that they clash with how they're written in stg_class_categories (INT64) and changing these to integer resolved that.

Comment (harjeetkalsi) [reply models/staging/stg_classes.sql]: Thanks, that makes sense. I agree the more important thing here is keeping linked key fields consistent across models. Since stg_class_categories is currently using integer-compatible IDs and is not being cast as anything explicitly, I’m happy with this change for now. In stg_class_categories.sql:

category_id is selected as-is
parent_category_id is cast to INT64
category_level is cast to INT64

So it looks like category_id may already be numeric in the raw source, which would explain why downstream matching expects integer-like types. 

It would just be good for us to agree a broader convention for IDs across the project so we stay consistent going forward.

**Review by harjeetkalsi — APPROVED** [review]

**Review by EmilyYoung26 — APPROVED** [review]
Looks nice and clean, happy to approve! Great job 😄
